neurodsp/sim/utils.py

```diff
@@ -0,0 +1,30 @@
+"""Utility functions for simulations."""
+
+###################################################################################################
+###################################################################################################
+
+def check_osc_def(n_seconds, fs, freq):
+    """Check whether a requested oscillation definition will have an integer number of cycles.
+
+    Parameters
+    ----------
+    n_seconds : float
+        Simulation time, in seconds.
+    fs : float
+        Signal sampling rate, in Hz.
+    freq : float
+        Oscillation frequency.
+
+    Returns
+    -------
+    bool
+        Whether the definition will have an integer number of cycles.
+    """
+
+    # Sampling rate check: check if the number of samples per cycle is an integer
+    srate_check = (fs/freq).is_integer()
+
+    # Time check: check if signal length matches an integer number of cycles
+    time_check = (n_seconds * freq).is_integer()
```

Comment:
Is truncation always an issue? This example seems to produce an appropriate spectrum:
```python
n_seconds, fs, freq = 1.99, 1000, 10

print((n_seconds * freq).is_integer())

sig = sim_oscillation(n_seconds, fs, freq)
freqs, powers = compute_spectrum(sig, fs)
plot_power_spectra(freqs, powers)
```

Reply:
huh, great catch! 

So yeh, this time check isn't super specific. Whether the number of cycles leads to funky PSD is a bit more complex than this check - it depends on the actual spectral estimation that is done (for example, with Welchs, I think it depends on the segment size (`nperseg`), and how this relates to the signal, more than the details of the total signal. Because of this, I think we should probably just entirely drop this time check, since I otherwise don't really know how to make it more specific (we can't predict the spectral estimation people will do - and this emphasizes that the details of the resulting PSD are somewhat contextual - not fully determined by the sim'd time series).